src/Provider/ImageProvider.php

```diff
@@ -83,25 +83,40 @@ public function getHelperProperties(MediaInterface $media, $format, $options = [
         ];
 
         if (MediaProviderInterface::FORMAT_ADMIN !== $format) {
-            $srcSet = [];
 
-            foreach ($this->getFormats() as $providerFormat => $settings) {
-                // Check if format belongs to the current media's context
-                if (0 === strpos($providerFormat, $media->getContext())) {
-                    $width = $this->resizer->getBox($media, $settings)->getWidth();
+            $srcSetFormats = $this->getFormats();
 
-                    $srcSet[] = sprintf('%s %dw', $this->generatePublicUrl($media, $providerFormat), $width);
+            if (isset($options['srcset']) && is_array($options['srcset'])) {
```

Comment:
Also, what if it isn't an array? This will continue happily, whereas it should fail as loudly and early as possible.

Reply:
Well, a string value was already supported and I didn’t want to introduce a BC break.

Reply:
Ooooh ok I get it!